### PR TITLE
Big Heretic revamp (finally)

### PR DESCRIPTION
**Pull request by ynot01**
# Document the changes in your pull request

Makes strong things weaker and weak things stronger.

# Changes

## Buff

- **Heretic's Base ash path now grants you 40% heat resistance**

*Guys gonna be on fire a lot. He should be better at that.*

## Buff

- **Heretic's upgraded ash blade applies more fire stacks on hit (1 => 2)**

*1 fire stack for this hefty investment is pretty pitiful, the other paths have serious debilitating effects on their upgraded blades (death chems, slash wounds), hopefully this is sufficient.*

## Nerf

- **Heretic's Ash Mark's detonate no longer applies stamina damage, but instead applies 2 fire stacks and ignites on hit**

*The typical combo for this was mansus grasp into blade, which resulted in an instant stamina crit, or a 10 second hardstun. This was not fun to fight against by any means. As a trade-off, it now applies some fire stacks on hit, so it isn't a boring damage ability.*

## Buff

- **Researching heretic influences takes less time (10 seconds => 4 seconds)**

*These obnoxiously long do_afters made it very hard to get on your feet during early game. Just very annoying and time-consuming, with a high chance of getting caught in more populated areas.*

## Trade-off

- **Drawing heretic runes takes less time (30 seconds => 8 seconds)**
- **You can now only have one heretic rune active at a time, drawing a new one will delete your old one**

*Same reasoning as above, but has additional balance aspects. If you could just slap down runes, they would be all over the place, and heretics could just go to any of their several runes whenever they wanted to do deeds. Here, it is a lot easier to slap down a rune so you don't have to drag your bodies a thousand miles or spend 30 seconds wondering if someone is going to catch you, but it is a bit of a double edged sword. Only having one rune active means security can track your activity or, by locating a rune, see where you've been recently.*

## Nerf

- **Breaking your heretic blade now takes 2 seconds of standing still**

*Another very unfun thing to play against. The current meta is to print out a thousand blades and just use it as an insta-warp whistle whenever you get in trouble, making you virtually uncatchable as long as you have blades to spare.*

## Buff

- **All heretic blades are now hooks that break shields if they don't get blocked by them**

*A new feature for heretics, a response to a critical lack of ability to deal with ranged combat. All blades can now be used at range, on a 3 second cooldown, to pull opponents into melee range. Doing so immobilizes them for a very short time, allowing for a free hit if you're quick. The shield thing is a little dash of spice to help heretics deal with shield spam.*

## Other stuff

- **Magical recharging wands and other magical ranged items no longer have one additional charge than they're supposed to**

*An oversight that was caused by wands charging endlessly, even if at max charge.*

# Changelog

:cl:  
tweak: Heretic's Base ash path now grants you 40% heat resistance
tweak: Heretic's upgraded ash blade applies more fire stacks on hit (1 => 2)
tweak: Heretic's Ash Mark's detonate no longer applies stamina damage, but instead applies 2 fire stacks and ignites on hit
tweak: Researching heretic influences takes less time (10 seconds => 4 seconds)
tweak: Drawing heretic runes takes less time (30 seconds => 8 seconds)
tweak: You can now only have one heretic rune active at a time, drawing a new one will delete your old one
tweak: Breaking your heretic blade now takes 2 seconds of standing still
tweak: All heretic blades are now hooks that break shields if they don't get blocked by them
bugfix: Magical recharging wands and other magical ranged items no longer have one additional charge than they're supposed to
/:cl:

### CODE REVIEW

**Comment by MajManatee** [comment]
this sounds pretty epic

**Comment by Skrem7** [comment]
Still more work to do on the antag as a whole but this is a good start that should help with straight up being a threat early. Next steps should probably be looking at flesh more keenly, then each of the non-path specific abilities that are otherwise obnoxious to use, maybe tuning them down but making them more viable (maybe less ingredient-heavy and doesn't require touch but still takes times regarding rituals, for example)

**Comment by Mantlecrawler** [comment]
as the notorious antag main (Heretic main) i approve of this PR

**Comment by MajManatee** [comment]
The T2 and T3 abilites are so much worse than the T1 abilities. They definetly need some improvements.

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm [hunk 1]
```diff
@@ -74,9 +75,10 @@
 	var/A = get_turf(target)
 	to_chat(user, span_danger("You start drawing a rune..."))
 
-	if(do_after(user, 30 SECONDS, A))
-
-		new /obj/effect/eldritch/big(A)
+	if(do_after(user, 8 SECONDS, A))
```
Comment (Moltijoe) [inline code/modules/antagonists/eldritch_cult/eldritch_book.dm:79]: you should document that it's reduced to 8 seconds, not 10 seconds

**Review by Moltijoe — APPROVED** [review]
Heretic has always been one of the cooler antags, good to see them getting love

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm (hunk 1 above)
Comment (MajManatee) [reply code/modules/antagonists/eldritch_cult/eldritch_book.dm:79]: even BETTER. 30 seconds was ABSURD, 10 even feels like a lot.

**Comment by RG4ORDR** [comment]
I am very interested to see where this will go.
Though this still doesn't solve a latent problem with a heretic getting arrested or random searched for being funny and having their heart and book on them.

**Comment by Burger1998** [comment]
Would it be possible to make it so ash path jaunt lets you actually move if your in stam crit? its very embarrassing jaunting and realizing its impossible to move and just popping back in front of the funny man with a batong

**Comment by kooarbiter** [comment]
this is genuinely a great change, especially giving heretics a ranged capability, I love it just because it speeds up runes and influences

**Comment by RG4ORDR** [comment]
> Would it be possible to make it so ash path jaunt lets you actually move if your in stam crit? its very embarrassing jaunting and realizing its impossible to move and just popping back in front of the funny man with a batong

No, this is one the funniest things I love to see.
Wizards,Vampires,Heretics getting fucked over by stamina trying to TP away is always funny.

**Comment by Moltijoe** [comment]
> No, this is one the funniest things I love to see. Wizards,Vampires,Heretics getting fucked over by stamina trying to TP away is always funny.

removing the stamina stun but keeping the stamina slow might be reasonable, so they can try to escape, but can realistically only make it like 4 tiles before reappearing.

**Comment by MajManatee** [comment]
they can ALREADY only move 4 tiles with the ash jaunt. Its really short range

**Comment by ThatLing** [comment]
Balance council said yes

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm [hunk 2]
```diff
@@ -9,6 +9,7 @@
 	var/mob/living/last_user
 	///Where we cannot create the rune?
 	var/static/list/blacklisted_turfs = typecacheof(list(/turf/closed,/turf/open/space,/turf/open/lava))
+	var/obj/effect/eldritch/big/last_rune
```
Comment (ThatLing) [inline code/modules/antagonists/eldritch_cult/eldritch_book.dm:12]: You probably need to qdel this one in /Destroy

**Review by ThatLing — CHANGES_REQUESTED** [review]

File: code/modules/antagonists/eldritch_cult/eldritch_book.dm [hunk 3]
```diff
@@ -9,6 +9,7 @@
...
 
 /obj/item/forbidden_book/Destroy()
 	last_user = null
```
Comment (ThatLing) [inline code/modules/antagonists/eldritch_cult/eldritch_book.dm:15]:
```suggestion
	last_user = null
	last_rune = null
```

**Review by ThatLing — APPROVED** [review]

**Pull request merged**